pkg/reconciler/buildrun/resources/taskrun.go

```diff
@@ -318,6 +341,23 @@ func GenerateTaskRun(
 	}
 
 	expectedTaskRun.Spec.Params = params
+
+	// Ensure a proper override of params between Build and BuildRun
+	// A BuildRun can override a param as long as it was defined in the Build
```

Comment:
I see the use case for this. 

For example:

> `a-build` doesnt define params, because the strategy parameters defaults are ok for the first associated buildRun. But it might happen that on subsequent buildRuns, the user will like to modify one of the strategy parameters values.

done.